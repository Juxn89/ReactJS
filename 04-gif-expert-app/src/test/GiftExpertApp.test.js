import { expect } from '@jest/globals';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';

import React from 'react';
import GiftExpertApp from '../giftExpertApp';

describe('Test on GiftExpertApp', () => {
    test('Should show it correctly ', () => {
        const wrapper = shallow(<GiftExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Should show a category list', () => {
        const categories = ['Samurai X', 'Makoto Shishio'];
        const wrapper = shallow(<GiftExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GiftGrid').length).toBe(categories.length);
    });
});