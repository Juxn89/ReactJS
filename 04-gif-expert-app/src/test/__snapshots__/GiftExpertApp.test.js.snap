// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GiftExpertApp Should show a category list 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GiftGrid
      category="Samurai X"
      key="Samurai X"
    />
    <GiftGrid
      category="Makoto Shishio"
      key="Makoto Shishio"
    />
  </ol>
</Fragment>
`;

exports[`Test on GiftExpertApp Should show it correctly  1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
